pkg/deploy.go

```diff
@@ -0,0 +1,189 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pkg
+
+import (
+	"fmt"
+	"github.com/submariner-io/submariner-operator/internal"
+	"github.com/submariner-io/submariner-operator/pkg/broker"
+
+	"github.com/submariner-io/admiral/pkg/stringset"
+	"github.com/submariner-io/submariner-operator/pkg/discovery/globalnet"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/utils"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/utils/restconfig"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/components"
+	v1 "k8s.io/api/core/v1"
+
+	submarinerv1a1 "github.com/submariner-io/submariner-operator/api/submariner/v1alpha1"
+	"github.com/submariner-io/submariner-operator/pkg/internal/cli"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/datafile"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/brokercr"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/submarinerop"
+)
+
+type DeployOptions struct {
+	IpsecSubmFile               string
+	GlobalnetEnable             bool
+	GlobalnetCIDRRange          string
+	DefaultGlobalnetClusterSize uint
+	ComponentArr                []string
+	GlobalCIDRConfigMap         *v1.ConfigMap
+	DefaultCustomDomains        []string
+	Repository                  string
+	ImageVersion                string
+	OperatorDebug               bool
+}
+
+var ValidComponents = []string{components.ServiceDiscovery, components.Connectivity}
+
+const brokerDetailsFilename = "broker-info.subm"
+
+func Deploy(do DeployOptions, kubeConfig string, kubeContext string) error {
+	fmt.Println("in broker.deploy, flags are %s", do)
+
+	status := cli.NewStatus()
```

Comment:
This assumes that `cli` is an appropriate status reporter, which isn’t a given; instead, `Deploy` should accept a status reporter as a parameter.